contracts/MuonNodeStaking.sol

```diff
@@ -0,0 +1,624 @@
+     * @param _muonAppId The Muon app ID.
+     * @param _muonPublicKey The Muon public key.
+     * @param bondedTokenAddress The address of the BondedToken contract.
+     */
+    function initialize(
+        address muonTokenAddress,
+        address nodeManagerAddress,
+        uint256 _muonAppId,
+        PublicKey memory _muonPublicKey,
+        address bondedTokenAddress
+    ) external initializer {
+        __MuonNodeStakingUpgradeable_init(
+            muonTokenAddress,
+            nodeManagerAddress,
+            _muonAppId,
+            _muonPublicKey,
+            bondedTokenAddress
+        );
+    }
+
+    function __MuonNodeStakingUpgradeable_init_unchained()
+        internal
+        initializer
+    {}
+
+    /**
+     * @dev Updates the list of staking tokens and their multipliers.
+     * Only callable by the DAO_ROLE.
+     * @param tokens The array of staking token addresses.
+     * @param multipliers The array of corresponding multipliers for each token.
+     */
+    function updateStakingTokens(
+        address[] calldata tokens,
+        uint256[] calldata multipliers
+    ) external onlyRole(DAO_ROLE) {
+        require(
+            tokens.length == multipliers.length,
+            "Mismatch in the length of arrays."
+        );
+
+        for (uint256 i = 0; i < tokens.length; i++) {
+            address token = tokens[i];
+            uint256 multiplier = multipliers[i];
+
+            if (isStakingToken[token] > 0) {
+                if (multiplier == 0) {
+                    uint16 tokenIndex = isStakingToken[token] - 1;
+                    address lastToken = stakingTokens[stakingTokens.length - 1];
+
+                    stakingTokens[tokenIndex] = lastToken;
+                    isStakingToken[lastToken] = isStakingToken[token];
+                    stakingTokens.pop();
+                    isStakingToken[token] = 0;
+                }
+
+                stakingTokensMultiplier[token] = multiplier;
+            } else {
+                require(
+                    multiplier > 0,
+                    "Invalid multiplier. The multiplier value must be greater than 0."
+                );
+                stakingTokens.push(token);
+                stakingTokensMultiplier[token] = multiplier;
+                isStakingToken[token] = uint16(stakingTokens.length);
+            }
+            emit StakingTokenUpdated(token, multiplier);
+        }
+    }
+
+    /**
+     * @dev Locks the specified tokens in the BondedToken contract for a given tokenId.
+     * The staker must first approve the contract to transfer the tokens on their behalf.
+     * Only the staker can call this function.
+     * @param tokenId The unique identifier of the token.
+     * @param tokens The array of token addresses to be locked.
+     * @param amounts The corresponding array of token amounts to be locked.
+     */
+    function lockToBondedToken(
+        uint256 tokenId,
```

Comment:
Why getting `tokenId` here?
`tokenId` is `users[msg.sender].tokenId`

and owner of the tokenId should be the staking contract.